src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -5146,28 +5146,31 @@ def create_service_account(
                 already exists.
         """
         with Session(self.engine) as session:
+            # Check if a service account with the given name already
+            # exists
+            err_msg = (
+                f"Unable to create service account with name "
+                f"'{service_account.name}': Found existing service "
+                "account with this name."
+            )
+            try:
+                self._get_account_schema(
+                    service_account.name, session=session, service_account=True
+                )
+                raise EntityExistsError(err_msg)
+            except KeyError:
+                pass
+
             # Create the service account
             new_account = UserSchema.from_service_account_request(
                 service_account
             )
             session.add(new_account)
-
-            # Check if a service account with the given name already
-            # exists
-            service_accounts = session.execute(
-                select(UserSchema).where(
-                    UserSchema.name == service_account.name,
-                    UserSchema.is_service_account.is_(True),  # type: ignore[attr-defined]
-                )
-            ).fetchall()
-            if len(service_accounts) == 1:
+            try:
                 session.commit()
-            else:
-                raise EntityExistsError(
-                    f"Unable to create service account with name "
-                    f"'{service_account.name}': Found existing service "
-                    "account with this name."
-                )
+            except IntegrityError as e:
+                raise EntityExistsError(err_msg) from e
```

Comment:
The thing about `IntegrityError` is you don't know for sure if it's raised because of a unique constraint is triggered in the DB. It could be other reasons, in which case this will report the wrong message. There are more ways to address this:

* just go with the flow. If it happens, it happens.
* find a way to determine if the integrity error really is raised because of a unique constraint violation (I really don't know how though, because parsing the error message is unreliable)
* raise this as a generic error instead of a specific one. For example, you could use `ValueError` with a message saying that an unexpected integrity error has been detected. It doesn't really matter what you use, as long as it's not RuntimeError and it's a generic error message that doesn't necessairly point to a unique constraint being detected (which you don't really know for sure happened)

Reply:
Agree, it might be of a different nature. I will let all commit-related exceptions bubble up freely as a safer option.